test/langtools/jdk/javadoc/doclet/testSnippetTag/TestSnippetTag.java

```diff
@@ -0,0 +1,2380 @@
+                    A.java:4: error: File not found: %s""".formatted(fileName));
+        checkNoCrashes();
+    }
+
+    @Test
+    public void testNoContents(Path base) throws Exception {
+        Path srcDir = base.resolve("src");
+        Path outDir = base.resolve("out");
+        new ClassBuilder(tb, "pkg.A")
+                .setComments("""
+                             {@snippet}
+                             """)
+                .setModifiers("public", "class")
+                .write(srcDir);
+        javadoc("-d", outDir.toString(),
+                "-sourcepath", srcDir.toString(),
+                "pkg");
+        checkExit(Exit.ERROR);
+        checkOutput(Output.OUT, true,
+                    """
+                    A.java:3: error: @snippet does not specify contents""");
+        checkNoCrashes();
+    }
+
+    @Test
+    public void testConflict20(Path base) throws Exception {
+        Path srcDir = base.resolve("src");
+        Path outDir = base.resolve("out");
+        new ClassBuilder(tb, "pkg.A")
+                .setComments("""
+                             {@snippet file="" class="" :
+                                 Hello, Snippet!
+                             }
+                             """)
+                .setModifiers("public", "class")
+                .write(srcDir);
+
+        javadoc("-d", outDir.toString(),
+                "-sourcepath", srcDir.toString(),
+                "pkg");
+        checkExit(Exit.ERROR);
+        // TODO
+        //   In this and all similar tests check that there are no other errors, let alone errors related to {@snippet}
+        //   To achieve that, we might need to change JavadocTester (i.e. add "consume output", "check that the output is empty", etc.)
+        checkOutput(Output.OUT, true,
+                    """
+                    A.java:3: error: @snippet specifies multiple external contents, which is ambiguous""");
+        checkNoCrashes();
+    }
+
+    @Test
+    public void testConflict30(Path base) throws Exception {
+        Path srcDir = base.resolve("src");
+        Path outDir = base.resolve("out");
+        new ClassBuilder(tb, "pkg.A")
+                .setComments("""
+                             {@snippet class="" file="" :
+                                 Hello, Snippet!
+                             }
+                             """)
+                .setModifiers("public", "class")
+                .write(srcDir);
+        javadoc("-d", outDir.toString(),
+                "-sourcepath", srcDir.toString(),
+                "pkg");
+        checkExit(Exit.ERROR);
+        checkOutputEither(Output.OUT,
+                          """
+                          A.java:3: error: @snippet specifies multiple external contents, which is ambiguous""");
+        checkNoCrashes();
+    }
+
+    // TODO: perhaps this method could be added to JavadocTester
+    private void checkOutputEither(Output out, String first, String... other) {
```

Comment:
1. This method allows a call with just a single String arg.
2. Given that the method allows many String args, the name might be better as `checkOneOf`

Note to self/@jonathan-gibbons: review all references for suggestions to JavadocTester